app/src/main/java/org/fossasia/susi/ai/activities/MainActivity.java

```diff
@@ -1417,18 +1431,77 @@ public boolean onOptionsItemSelected(MenuItem item) {
+                                    String.format(getString(R.string.promo_msg_template),
+                                            String.format(getString(R.string.app_share_url),getPackageName())));
+                            startActivity(shareIntent);
+                        }
+                        catch (Exception e) {
+                            Snackbar.make(coordinatorLayout, getString(R.string.error_msg_retry), Snackbar.LENGTH_SHORT).show();
+                        }
+                    }
+                });
+                facebookInvite.setOnClickListener(new View.OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        BranchUniversalObject branchUniversalObject = new BranchUniversalObject()
+                                .setCanonicalIdentifier("user/" + Settings.Secure.getString(getContentResolver(), Settings.Secure.ANDROID_ID))
+                                .setTitle(getString(R.string.invite_title))
+                                .setContentDescription(getString(R.string.invite_content_description))
+                                .setContentImageUrl(getString(R.string.share_fb_preview_url))
+                                .addContentMetadata("userId", Settings.Secure.getString(getContentResolver(), Settings.Secure.ANDROID_ID))
+                                .addContentMetadata("packageName", getPackageName());
+
+                        LinkProperties linkProperties = new LinkProperties()
+                                .setChannel(getString(R.string.facebook))
+                                .setFeature(getString(R.string.invite_via_facebook));
+
+                        branchUniversalObject.generateShortUrl(MainActivity.this, linkProperties, new Branch.BranchLinkCreateListener() {
+                            @Override
+                            public void onLinkCreate(String url, BranchError error) {
+                                if (error == null && AppInviteDialog.canShow()) {
+                                    AppInviteContent content = new AppInviteContent.Builder()
+                                            .setApplinkUrl(url)
+                                            .setPreviewImageUrl(getString(R.string.share_fb_preview_url))
+                                            .build();
+                                    AppInviteDialog appInviteDialog = new AppInviteDialog(MainActivity.this);
+                                    appInviteDialog.registerCallback(callbackManager, new FacebookCallback<AppInviteDialog.Result>() {
+                                        @Override
+                                        public void onSuccess(AppInviteDialog.Result result) {
+                                            alertDialog.cancel();
+                                        }
+
+                                        @Override
+                                        public void onCancel() {
```

Comment:
@jig08 @the-dagger @daminisatya  should the method be left empty or should I include a Log to pass Codacy?